vnext/ReactUWP/Views/TouchEventHandler.cpp

```diff
@@ -427,6 +446,42 @@ bool TouchEventHandler::TagFromOriginalSource(
   return true;
 }
 
+winrt::IPropertyValue TouchEventHandler::TestHit(
+    const winrt::Collections::IVectorView<xaml::Documents::Inline> &inlines,
+    const winrt::Point &pointerPos) {
+  winrt::IPropertyValue tag(nullptr);
+  for (const auto &el : inlines) {
+    if (const auto span = el.try_as<winrt::Documents::Span>()) {
+      const auto resTag = TestHit(span.Inlines().GetView(), pointerPos);
+
+      if (resTag) {
+        return resTag;
+      }
+
+    } else if (const auto run = el.try_as<winrt::Documents::Run>()) {
+      const auto start = el.ContentStart();
+      const auto end = el.ContentEnd();
+
+      const auto startRect = start.GetCharacterRect(winrt::Documents::LogicalDirection::Forward);
+      const auto endRect = end.GetCharacterRect(winrt::Documents::LogicalDirection::Forward);
+
+      // Approximate the bounding rect (for now, don't account for text wrapping or right-to-left orientation)
+      if ((startRect.X <= pointerPos.X) && (endRect.X + endRect.Width >= pointerPos.X) &&
```

Comment:
I actually think you can simplify this a lot:

```cpp
auto unionRect = startRect.Union(endRect);
if (unionRect.Contains(pointerPos)) {
...
}
```

Reply:
This is a good suggestions. However, when I wrote this:
```
auto unionRect = winrt::Windows::UI::Xaml::RectHelper::Union(startRect, endRect);
if (winrt::Windows::UI::Xaml::RectHelper::Contains(unionRect, pointerPos)) {
```
I started getting strange rectangles, and the thing stopped working - clicking on some nested text would increment a counter from the level above, if it happened towards the left of the line.